common/message.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview Types regarding communication between
+ * diagostics-app and diagnostics-extension
+ */
+
+export interface Request {
+  scope: RequestScope;
+  category: RequestCategory;
+}
```

Comment:
I think "category" sounds generic, however, it is being used for telemetry requests only (probably?). I think the following might be better:
export interface Request {
  type: RequestType; // instead of RequestScope
  telemetry?: { // optional
    infoType: TelemetryInfoType // instead of TelemetryCategory
  },
  diagnostics?: {}, // optional
  event?: { ... } // optional
}

telemetry, diagnostics, event, are all optional. This way, your backend code can process more than one request simultaneously (if needed): both telemetry and event for example. A request missing any of those optional fields is considered incorrect, ignore it.

The idea is to use a request structure that can be extended without breaking existing code parsing the request.

Reply:
You may come up with a better proposal. Let me know.

Reply:
actually if we need to support more than one request simultaneously, then we can omit the "type" field.

Also, if you decide to support one request type, you can support an optional Array of "telemetry" request, so that we can retrieve multiple TelemetryInfo at once.

Up to you.

Reply:
Yes, this looks much better. 👍🏻 

I don't think we will need to support more than one request simultaneously.

1. Telemetry data are requested by components individually as per their respective refresh rates, the only time when we will need to club them is by supporting batch requests from the frontend (which won't be required since we're doing this inside chrome environment)

2. When a diagnostics page is open, we won't need telemetry/event details, so that need to be sent individually.

3. Events are just a long lived connection which will be handled separately once an initial handshake is done between extension and the app for starting this connection.

The schema that you sent sounds perfect. :+1: